core/dbt/graph/selector_methods.py

```diff
@@ -143,6 +144,15 @@ def non_source_nodes(
             self.metric_nodes(included_nodes),
         )
 
+    def groupable_nodes(
+        self,
+        included_nodes: Set[UniqueId],
+    ) -> Iterator[Tuple[UniqueId, Union[ManifestNode, Metric]]]:
+        yield from chain(
+            self.parsed_nodes(included_nodes),
+            self.metric_nodes(included_nodes),
```

Comment:
Metric nodes don't have a "group" field.

Reply:
Metric nodes should have a "group" field as per https://github.com/dbt-labs/dbt-core/pull/7069#discussion_r1119179657. Thank you @gshank for fixing that as part of: https://github.com/dbt-labs/dbt-core/pull/7069!

Reply:
I've added it in my pull request. I guess it doesn't have to be handled in this pr.